node/node.go

```diff
@@ -298,6 +268,10 @@ func optionallyCreateAndStartPrivValidator(config *cfg.Config, privValidator typ
 	return privValidator, nil
 }
 
+// This returns whether or not fast sync is enabled based on (1) whether
+// `config.FastSync` is true, and (2) if there's only a single validator,
+// whether the given `privValidator`'s address is not the same as the sole
+// validator's address.
 func optionallyFastSync(config *cfg.Config, state sm.State, privValidator types.PrivValidator) bool {
```

Comment:
```go
func onlyValidatorIsUs(state sm.State, privVal types.PrivValidator) bool {
	if state.Validators.Size() > 1 {
		return false
	}

	addr, _ := state.Validators.GetByIndex(0)
	return bytes.Equal(privVal.GetPubKey().Address(), addr)
}
```

Here is what is bad about optionallyFastSync func:

1) `optionallyFastSync` suggest that the function performs fast sync, which it does not
2) even if we rename it to `onlyValidatorIsUs(config.FastSync, ...) (fastSync bool)`, we'll be mixing responsibilities => spaghetti code. function should have only one primary responsibility

I think we should extract onlyValidatorIsUs func and use it to make a decision if we should fast sync or not inside another function or directly in NewNode func

```go
fastSync := config.FastSync && !onlyValidatorIsUs(state, privValidator)
```

Reply:
Sure, we can refine it this way if you want.